### PR TITLE
fix(experiments): render LoadingState when changing experiment goal

### PR DESCRIPTION
**Pull request by jurajmajerik**
## Problem
Fixes https://posthoghelp.zendesk.com/agent/tickets/9620 (moved to PostHog: https://us.posthog.com/project/2/support/tickets/13151)

The issue wasn't that the results were not refreshing, but that the loader was not showing. Because it took a long time for Raquel's results to load, it looked as though the results were not refreshing.

This was happening because we were showing the loader based on whether `experimentResults` existed. This value doesn't exist on the initial load but is always present thereafter.

## Changes
Render `LoadingState` at the top level if `experimentResultsLoading` is true.

Before:
https://github.com/PostHog/posthog/assets/22996112/43b747b1-69b2-45a2-99cf-0b227c5fc079

After:
https://github.com/PostHog/posthog/assets/22996112/f1e84732-b15f-4fd9-8736-8a898f3b54cf

This code could be cleaned up further, but might as well leave it for the UI revamp :)

## How did you test this code?
👀

### CODE REVIEW

**Comment by jurajmajerik** [comment]
github not embedding the videos for some reason 🤔

File: frontend/src/scenes/experiments/ExperimentResult.tsx [hunk 1]
```diff
@@ -37,181 +37,197 @@ export function ExperimentResult(): JSX.Element {
...
-                                        <LemonProgress
-                                            percent={Number((experimentResults.probability[variant] * 100).toFixed(1))}
-                                            strokeColor={getSeriesColor(
-                                                getIndexForVariant(variant, experimentInsightType)
+                                        ))
+                                }
+                            </div>
+                        )
+                    ) : (
+                        !experimentResults && experimentResultsLoading && <LoadingState />
```
Comment (neilkakkar) [inline frontend/src/scenes/experiments/ExperimentResult.tsx:176]: github diffs make this hard to read, but do we need this here at all, given you now have it at the top level on L40-45 ?

File: frontend/src/scenes/experiments/ExperimentResult.tsx [hunk 2]
```diff
@@ -37,181 +37,197 @@ export function ExperimentResult(): JSX.Element {
...
+                    )}
+                    {experimentResults && !experimentResultsLoading ? (
```
Comment (neilkakkar) [inline frontend/src/scenes/experiments/ExperimentResult.tsx:178]: github diffs make this hard to read, but do we need this here at all, given you now have it at the top level on L40-45 ?

Comment (jurajmajerik) [reply frontend/src/scenes/experiments/ExperimentResult.tsx:175]: indeed, removed 👍

Comment (neilkakkar) [reply frontend/src/scenes/experiments/ExperimentResult.tsx:175]: @jurajmajerik just circling back on this, this seems to still be here? 😅 (there were 2 places)

Comment (neilkakkar) [reply frontend/src/scenes/experiments/ExperimentResult.tsx:175]: oh lol I see conflicts with my PR, I'll do it 👍

**Pull request merged**